app/scripts/controllers/transactions/index.js

```diff
@@ -672,27 +672,36 @@ export default class TransactionController extends EventEmitter {
    * state is unapproved. Returns the updated transaction.
    *
    * @param {string} txId - transaction id
+   * @param {number} currentSendFlowHistoryLength - sendFlowHistory entries currently
    * @param {Array<{ entry: string, timestamp: number }>} sendFlowHistory -
    *  history to add to the sendFlowHistory property of txMeta.
    * @returns {TransactionMeta} the txMeta of the updated transaction
    */
-  updateTransactionSendFlowHistory(txId, sendFlowHistory) {
+  updateTransactionSendFlowHistory(
+    txId,
+    currentSendFlowHistoryLength,
```

Comment:
could we preserve the name of the value here? 
Instead of `currentSendFlowHistoryLength` ->   `unapprovedTxSendFlowHistoryLength`? 

The reason I ask is that at first... I was thinking.. we are appending the flow history onto the current one, so why check these lengths? And then I traced through the code to understand it better. I think it would be helpful to update this name.

Reply:
The method is generic to update transactionSendFlowHistory not specifically for unapproved transaction.

Reply:
I think, actually, I was confused about how this function worked. After reading https://github.com/MetaMask/metamask-extension/pull/15585/files#r963061208, it clicked. Thanks @brad-decker! 

@jpuri your update to change it to `currentSendFlowHistoryLength` sounds good to me now. Thank you and thanks for this explanation too 🙏